tools/serverpod_cli/lib/src/analyzer/models/validation/restrictions/default.dart

```diff
@@ -208,4 +212,94 @@ class DefaultValueRestriction extends ValueRestriction {
 
     return errors;
   }
+
+  List<SourceSpanSeverityException> _uuidValueValidation(
+    dynamic value,
+    SourceSpan? span,
+  ) {
+    var errors = <SourceSpanSeverityException>[];
+
+    if (value is UuidValue) {
+      try {
+        value.validate();
+        return [];
+      } catch (e) {
+        errors.add(
+          SourceSpanSeverityException(
+            'The "$key" contains an invalid UUID value.',
+            span,
+          ),
+        );
+        return errors;
+      }
+    }
+
+    String invalidValueError =
+        'The "$key" value must be a "random" or valid UUID string (e.g., "$key"=random or "$key"=\'550e8400-e29b-41d4-a716-446655440000\').';
+
+    if (value is! String || value.isEmpty) {
+      errors.add(
+        SourceSpanSeverityException(
+          invalidValueError,
+          span,
+        ),
+      );
+      return errors;
+    }
+
+    bool invalidDefaultValue = value != defaultUuidValueRandom &&
+        !value.startsWith("'") &&
+        !value.startsWith('"');
+
+    if (invalidDefaultValue) {
+      errors.add(
+        SourceSpanSeverityException(
+          invalidValueError,
+          span,
+        ),
+      );
+      return errors;
+    }
+
+    if (value == defaultUuidValueRandom) return [];
+
+    bool validSingleQuote = isValidSingleQuote(value);
+    bool validDoubleQuote = isValidDoubleQuote(value);
+
+    if (value.startsWith("'") && !validSingleQuote) {
+      errors.add(
+        SourceSpanSeverityException(
+          'The "$key" must be a quoted string (e.g., "$key"=\'550e8400-e29b-41d4-a716-446655440000\').',
+          span,
+        ),
+      );
+      return errors;
+    } else if (value.startsWith('"') && !validDoubleQuote) {
+      errors.add(
+        SourceSpanSeverityException(
+          'The "$key" must be a quoted string (e.g., "$key"="550e8400-e29b-41d4-a716-446655440000").',
+          span,
+        ),
+      );
+      return errors;
+    }
+
+    /// Extract the actual UUID string by removing quotes
+    String uuidString = value.substring(1, value.length - 1);
+
+    bool isValidUuid = RegExp(
```

Comment:
Suggestion: Use UUID v4 validation here instead of your defined regexp.

It should be possible to just parse it here.